driver/src/main/com/mongodb/DBObjectCodec.java

```diff
@@ -72,7 +72,7 @@
     static BsonTypeClassMap createDefaultBsonTypeClassMap() {
         Map<BsonType, Class<?>> replacements = new HashMap<BsonType, Class<?>>();
         replacements.put(BsonType.REGULAR_EXPRESSION, Pattern.class);
-        replacements.put(BsonType.SYMBOL, String.class);
+        replacements.put(BsonType.SYMBOL, Symbol.class);
```

Comment:
While it may be more correct, this is a behavior change from 2.x, which decodes symbols as strings.  So the code here is as intended.  To fix the bug, I'd relax StringCodec to allow it to decode a symbol as a string.